File: src/main/java/com/moira/itda/domain/login/mapper/LoginMapper.java
package com.moira.itda.domain.login.mapper;

import com.moira.itda.global.entity.LoginHistory;
import com.moira.itda.global.entity.User;
import org.apache.ibatis.annotations.Mapper;

@Mapper
public interface LoginMapper {
    /**
     * 로그인 > 이메일로 User 조회
     */
    User selectUserByEmail(String email);

    /**
     * 로그인 > RTK 저장
     */
    void updateUserWhenLogin(String userId, String rtk);

    /**
     * 로그인 > 로그인 기록 저장
     */
    void insertLoginHistory(LoginHistory loginHistory);

    /**
     * 로그아웃 > RTK 삭제
     */
    void updateUserWhenLogout(String userId);
}

File: src/main/kotlin/com/moira/itda/domain/account/login/mapper/LoginMapper.kt
package com.moira.itda.domain.account.login.mapper

import com.moira.itda.global.entity.User
import com.moira.itda.global.entity.UserLoginHistory
import org.apache.ibatis.annotations.Mapper
import java.time.ZonedDateTime

@Mapper
interface LoginMapper {
    /**
     * 로그인 > User 조회
     */
    fun selectUserByEmail(email: String): User?

    /**
     * 로그인 > 계정 정지 기한 조회
     */
    fun selectBannedUntil(userId: String): ZonedDateTime

    /**
     * 로그인 > RefreshToken 수정
     */
    fun updateRefreshToken(userId: String, rtk: String)

    /**
     * 로그인 > LoginHistory 저장
     */
    fun insertUserLoginHistory(userLoginHistory: UserLoginHistory)
}